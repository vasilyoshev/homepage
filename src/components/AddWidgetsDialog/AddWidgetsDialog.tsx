import {
  Dialog,
  DialogContent,
  DialogTitle,
  IconButton,
  Typography,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import React from 'react';
import './AddWidgetsDialog.scss';

type AddWidgetsDialogType = {
  isOpen: boolean;
  onClose: () => void;
};

export const AddWidgetsDialog: React.FC<AddWidgetsDialogType> = ({
  isOpen,
  onClose,
}) => {
  return (
    <Dialog onClose={onClose} open={isOpen}>
      <DialogTitle disableTypography className="add-widget-dialog-title">
        <Typography variant="h4">Add Widgets</Typography>
        <IconButton onClick={onClose}>
          <CloseIcon />
        </IconButton>
      </DialogTitle>
      <DialogContent>Test</DialogContent>
    </Dialog>
  );
};
