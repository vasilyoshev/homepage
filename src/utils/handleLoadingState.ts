import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
