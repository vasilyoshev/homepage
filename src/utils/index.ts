export * from './handleLoadingState';
export * from './testUtils';
