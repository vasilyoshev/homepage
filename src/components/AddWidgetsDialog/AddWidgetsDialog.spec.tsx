import { render } from 'utils';
import { AddWidgetsDialog } from './AddWidgetsDialog';

describe('AddWidgetsDialog', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<AddWidgetsDialog isOpen={true} onClose={() => null} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
