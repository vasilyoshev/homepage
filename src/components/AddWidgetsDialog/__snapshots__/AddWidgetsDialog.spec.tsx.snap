// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddWidgetsDialog renders successfully 1`] = `<DocumentFragment />`;
